frame/collective/src/lib.rs

```diff
@@ -258,23 +269,21 @@ pub mod pallet {
 	#[pallet::origin]
 	pub type Origin<T, I = ()> = RawOrigin<<T as frame_system::Config>::AccountId, I>;
 
-	/// The hashes of the active proposals.
-	#[pallet::storage]
-	#[pallet::getter(fn proposals)]
-	pub type Proposals<T: Config<I>, I: 'static = ()> =
-		StorageValue<_, BoundedVec<T::Hash, T::MaxProposals>, ValueQuery>;
-
 	/// Actual proposal for a given hash, if it's current.
 	#[pallet::storage]
-	#[pallet::getter(fn proposal_of)]
 	pub type ProposalOf<T: Config<I>, I: 'static = ()> =
-		StorageMap<_, Identity, T::Hash, <T as Config<I>>::Proposal, OptionQuery>;
+		CountedStorageMap<_, Identity, T::Hash, BoundedVec<u8, ConstU32<MAX_SIZE>>, OptionQuery>;
```

Comment:
Sorry if this was a missunderstanding. We still need to use the preimage pallet, otherwise it is a re-implementation of it https://github.com/paritytech/substrate/issues/12070. This is not really an unbound lookup, but it wastes massive PoV since the benchmarking will *always* assume the worst-case of 4 MiB.    
Anyway I think you deserve a tip for this already, even if it does not merge 🙈 @juangirini  

I think this map should be `ProposalHash -> Bounded<Proposal>`. Does it make sense?

Reply:
@muraca if you want to be tipped please add your Kusaman or Polkadot address in the PR comment with this format
"[Polkadot | Kusama] address: xxxxx"

Reply:
> the benchmarking will always assume the worst-case of 4 MiB.

@ggwpez I took a look at the benchmarks and weight functions before implementing this, and I saw that weights depend on the length bound passed as input parameter in the extrinsics.
If you are positive that this wouldn't work with the PoV size, or that even if it's bounded we don't like to re-implement preimages, a thumbs up is enough for me to proceed and re-add preimages.

I really really dislike the `ProposalHash -> Bounded<Proposal>` approach, I feel it just wastes storage space. Alternatively, I could use the `note` method to store the preimage and generate a hash to put in a `Bounded::Lookup` variant when I need to interact with preimages to request or delete from storage.